.github/workflows/snyk-sbom.yml

```diff
@@ -0,0 +1,25 @@
+name: Snyk - Software Bill of Materials (SBOM)
+on:
+  release:
+    types: [published]
+
+jobs:
+  security:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      - name: Generate SBOM
+        uses: snyk/actions/node@master
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+        with:
+          command: sbom
+          args: --format=cyclonedx1.5+json --all-projects > sbom.json
+      - name: Upload SBOM
+        uses: Shopify/upload-to-release@c77c9b3e5d288adaef98a7007bf92340ec6ce03b # v2.0.0
+        with:
+          name: sbom.json
+          path: sbom.json
+          repo-token: ${{ secrets.GH_TOKEN }}
```

Comment:
Github Actions has automatic token authentication. There is not need to introduce a new PAT. Generally we try to avoid introducing PATs for production workloads, since they are inherently problematic 

- Bind to a user 
- Non centrally managed 
- Expiration dates not policied 
- Can be compromised 

@enzowritescode  Give this a read and let me know if you encounter any issues https://docs.github.com/en/actions/security-guides/automatic-token-authentication
🚀